apps/ledger-live-desktop/src/renderer/families/evm/SendAmountFields/SelectFeeStrategy.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { memo } from "react";
+import styled from "styled-components";
+import {
+  getAccountCurrency,
+  getAccountUnit,
+  getMainAccount,
+} from "@ledgerhq/live-common/account/index";
+import { Trans } from "react-i18next";
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import { GasOptions, Transaction as EvmTransaction } from "@ledgerhq/coin-evm/types";
+import { getTypedTransaction } from "@ledgerhq/coin-evm/transaction";
+import TachometerMedium from "~/renderer/icons/TachometerMedium";
+import CounterValue from "~/renderer/components/CounterValue";
+import FormattedVal from "~/renderer/components/FormattedVal";
+import TachometerHigh from "~/renderer/icons/TachometerHigh";
+import TachometerLow from "~/renderer/icons/TachometerLow";
+import Box, { Tabbable } from "~/renderer/components/Box";
+import Text from "~/renderer/components/Text";
+import Clock from "~/renderer/icons/Clock";
+import { getEstimatedFees } from "@ledgerhq/coin-evm/logic";
+
+type OnClickType = {
+  feesStrategy: keyof GasOptions;
+};
+
+type Props = {
+  onClick: (arg: OnClickType) => void;
+  transaction: EvmTransaction;
+  account: AccountLike;
+  parentAccount: Account | null | undefined;
+  gasOptions: GasOptions;
+};
+
+const FeesWrapper = styled(Tabbable)`
+  flex-direction: column;
+  align-items: center;
+  justify-content: space-between;
+  gap: 6px;
+
+  border: ${p =>
+    `1px solid ${
+      p?.selected ? p.theme.colors.palette.primary.main : p.theme.colors.palette.divider
+    }`};
+  padding: 20px 16px;
+  font-family: "Inter";
+  border-radius: 4px;
+  width: 140px;
+
+  &:hover {
+    cursor: "pointer";
+  }
+`;
+
+const FeesHeader = styled(Box)<{ selected?: boolean }>`
+  color: ${p =>
+    p.selected ? p.theme.colors.palette.primary.main : p.theme.colors.palette.text.shade50};
+`;
+
+const FeesValue = styled(Box)`
+  flex-direction: column;
+  align-items: center;
+  text-align: center;
+`;
+
+const ApproximateTransactionTime = styled(Box)<{ selected?: boolean }>`
+  flex-direction: row;
+  align-items: center;
+  border-radius: 3px;
+  background-color: ${p =>
+    p.selected ? p.theme.colors.palette.primary.main : p.theme.colors.palette.text.shade20};
+  padding: 5px 6px;
+`;
+
+const strategies: (keyof GasOptions)[] = ["slow", "medium", "fast"];
+
+const SelectFeeStrategy = ({ transaction, account, parentAccount, onClick, gasOptions }: Props) => {
+  const mainAccount = getMainAccount(account, parentAccount);
+  const accountUnit = getAccountUnit(mainAccount);
+  const feesCurrency = getAccountCurrency(mainAccount);
+
+  return (
+    <Box horizontal justifyContent="center" flexWrap="wrap" gap="16px">
+      {strategies.map(strategy => {
+        const selected = transaction.feesStrategy === strategy;
+
+        const gasOption = gasOptions[strategy];
+
+        const estimatedFees = getEstimatedFees(getTypedTransaction(transaction, gasOption));
```

Comment:
Maybe a `useMemo` just to maintain the ref of this variable ?

Reply:
Not possible here since this `estimatedFees` is defined inside a callback and for each strategy in `strategies`. Another possibility would be to manually and individually define an estimated fee value for each strategy before the return statement and use a switch case or if cascade inside the callback to select the correct estimated fee value every time. Not really worth the added complexity and reduced readability IMHO.

Reply:
Nevermind just realised that it's not helping for re-renders here, the solution would be to memoize directly the result for each strategy, either with a subcomponent or with a `useMemo` of the expected JSX for each `gasOption` 🤔